src/components/organisms/NavBarSchedule/NavBarSchedule.tsx

```diff
@@ -97,21 +110,88 @@ export const NavBarSchedule: React.FC<NavBarScheduleProps> = ({
     toggle: togglePersonalisedSchedule,
   } = useShowHide(false);
 
+  const liveAndFutureEvents = useMemo(
+    () =>
+      relatedVenueEvents.filter(
+        (event) =>
+          secondsToMilliseconds(
+            event.start_utc_seconds + minutesToSeconds(event.duration_minutes)
+          ) > startOfToday().getTime()
+      ),
+    [relatedVenueEvents]
+  );
+
+  const liveEventsMinimalStartValue = Math.min(
+    ...liveAndFutureEvents.map((event) => event.start_utc_seconds)
+  );
+
+  const firstLiveEvent = liveAndFutureEvents.find(
+    (event) => event.start_utc_seconds === liveEventsMinimalStartValue
+  );
+
+  const minDate = useMemo(
+    () =>
+      firstLiveEvent
+        ? isEventLiveOrFuture(firstLiveEvent)
+          ? new Date(liveEventsMinimalStartValue).getTime()
+          : millisecondsToSeconds(todaysDateTime)
+        : minRangeValue,
+    [firstLiveEvent, liveEventsMinimalStartValue]
+  );
+
+  const firstRangeDayTimeValue = isDateLessOrEqualsToday({
+    valueSource: secondsToMilliseconds(minDate),
+    valueTarget: millisecondsToSeconds(startOfToday().getTime()),
+  })
+    ? millisecondsToSeconds(todaysDate.getTime())
+    : millisecondsToSeconds(new Date(secondsToMilliseconds(minDate)).getTime());
+
+  const maxDate = useMemo(
+    () =>
+      Math.max(
+        ...liveAndFutureEvents.map(
+          (event) =>
+            event.start_utc_seconds + minutesToSeconds(event.duration_minutes)
+        ),
+        firstRangeDayTimeValue + 1
+      ),
+    [liveAndFutureEvents, firstRangeDayTimeValue]
+  );
+
+  const daysInBetween = differenceInDays(
+    fromUnixTime(maxDate),
+    fromUnixTime(firstRangeDayTimeValue)
+  );
+  const dayDifference = getEventDayRange(daysInBetween);
+
   const weekdays = useMemo(() => {
     const formatDayLabel = (day: Date | number) => {
       if (isScheduleTimeshifted) {
-        return format(day, "E, LLL d");
+        return format(day, "do");
       } else {
         return formatDateRelativeToNow(day, {
-          formatOtherDate: (dateOrTimestamp) => format(dateOrTimestamp, "E"),
+          formatOtherDate: (dateOrTimestamp) => format(dateOrTimestamp, "do"),
+          formatTomorrow: (dateOrTimestamp) => format(dateOrTimestamp, "do"),
         });
       }
     };
 
-    return range(SCHEDULE_SHOW_DAYS_AHEAD).map((dayIndex) => {
-      const day = addDays(firstDayOfSchedule, dayIndex);
+    return range(dayDifference).map((dayIndex) => {
+      const firstScheduleDate = isDateLessOrEqualsToday({
+        valueSource: secondsToMilliseconds(minDate),
+        valueTarget: millisecondsToSeconds(startOfToday().getTime()),
+      })
+        ? todaysDate
+        : new Date(secondsToMilliseconds(minDate));
```

Comment:
looks like we do pretty same calculations for `firstRangeDatimeValue`. I think we can reuse one of them.

Reply:
yep, fixed b41ef61